apps/browser/qml/pages/components/TabView.qml

```diff
@@ -203,10 +200,84 @@ Item {
 
         ]
     }
+
     TabsToolBar {
         id: tabsToolBar
         anchors.bottom: parent.bottom
         onBack: pageStack.pop()
-        onEnterNewTabUrl: root.enterNewTabUrl()
+        onEnterNewTabUrl: tabView.enterNewTabUrl()
+        onOpenMenu: popupMenu.active = true
+    }
+
+    PopUpMenu {
+        id: popupMenu
+
+        signal closeAllTabs
+
+        anchors.fill: parent
+        active: false
+
+        menuItem: Component {
+            Item {
+                id: menuItem_
+                readonly property int iconWidth: Theme.iconSizeMedium + Theme.paddingLarge
+                readonly property int verticalPadding: 3 * Theme.paddingSmall
+                height: content.height + verticalPadding * 2
+
+                Column {
+                    id: content
+                    width: parent.width
+                    spacing: Theme.paddingLarge
+                    y: menuItem_.verticalPadding
+
+                    Column {
+                        width: parent.width
+
+                        OverlayListItem {
+                            height: Theme.itemSizeSmall
+                            iconWidth: menuItem_.iconWidth
+                            iconSource: "image://theme/icon-m-tab-new"
+                            //% "New tab"
+                            text: qsTrId("sailfish_browser-la-new_tab")
+                            onClicked: {
+                                tabView.privateMode = false
+                                tabView.enterNewTabUrl()
+                            }
+                        }
+
+                        OverlayListItem {
+                            height: Theme.itemSizeSmall
+                            iconWidth: menuItem_.iconWidth
+                            iconSource: "image://theme/icon-m-incognito-new"
+                            //% "New private tab"
+                            text: qsTrId("sailfish_browser-la-new_private_tab")
+                            onClicked: {
+                                tabView.privateMode = true
```

Comment:
Because of the switch between modes, and the page sliding back, there can be a strange sliding effect if "New private tab" is selected while in normal mode, or vice versa (the tabs slide in one direction while the page slides in the other). Could the mode switch be delayed to avoid this (or something similar)? I appreciate this may not be straightforward to implement, so just a suggestion.

Reply:
The effect arises from the the (ab)use of Background in the popup menu to round the menu corners. I might have to have another go at decoupling the texture coordinates from the screen coordinates so that page movement isn't an issue.

It would still be a good idea to delay any action that ends up animating a page until after the menu closes. Or delay closing the menu until after the page is off screen as concurrent animations often end up looking messy.

Reply:
Ah, actually I wasn't referring to this effect. I had something simpler in mind: if the tab switches, while the page pops, then the tab pane is sliding from left to right (or right to left). At the same time the page is sliding right (back up the page stack). The result is that the tab ends up moving either double speed (right + right = double speed), or appears roughly static (left + right = static). It's not quite that simple because the transitions aren't linear, but hopefully you get the idea. The combined effect looks a little odd in my opinion, but it doesn't look broken.

What you describe also sounds like an important issue, but I'd like to distinguish the two to avoid confusion.

Reply:
dropped binding for TabView. done

Reply:
The animations in and out of the tab view while also switching to a different mode (private/normal) is much cleaner now. Thank you.